src/pyedb/dotnet/edb_core/cell/layout.py

```diff
@@ -331,3 +333,21 @@ def find_component_by_name(self, value: str):
         """
         obj = self._pedb._edb.Cell.Hierarchy.Component.FindByName(self._edb_object, value)
         return EDBComponent(self._pedb, obj) if obj is not None else None
+
+    def find_primitive(self, layer_name: Union[str, list]) -> list:
+        """Find a primitive objects by layer name.
+
+        Parameters
+        ----------
+        layer_name : str, list
+            Name of the layer.
+        Returns
+        -------
+        list
+        """
+        layer_name = layer_name if isinstance(layer_name, list) else [layer_name]
+        prims = []
+        for i in self.primitives:
+            if i.layer_name in layer_name:
+                prims.append(i)
+        return prims
```

Comment:
I would use list comprehension.
return [prim for prim in self.primitives if prim.layer_name == layer_name]